### PR TITLE
Fix clean demux permission error and query filter

### PR DESCRIPTION
**Pull request by barrystokman**
## Description
HK tabels Files hast fastq files that are not in demultiplexed-runs. Often they are in personal directories. Not only are thsoe files irrelevant, they also lead to `PermissionError` exceptions.

### Fixed
Only get HK files that are in `/home/proj/<env>/demultiplexed-runs/`
Handle permission errors

### How to prepare for test
- [ ] ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] paxa the environment: `paxa`
- [ ] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/clean/demultiplexed_flow_cells.py [hunk 1]
```diff
@@ -136,13 +148,11 @@ def files_exist_on_disk(self) -> bool:
                 )
             else:
                 if self.fastq_files_exist_in_housekeeper:
-                    self._files_exist_on_disk: bool = all(
-                        [Path(fastq_file.path).exists() for fastq_file in self.hk_fastq_files]
-                    )
+                    fastq_files_exist_on_disk = self._check_files_existence(self.hk_fastq_files)
```
Comment (henrikstranneheim) [inline cg/meta/clean/demultiplexed_flow_cells.py:151]:
```suggestion
                    fastq_files_exist_on_disk: List[bool] = self._check_files_existence(self.hk_fastq_files)
```

File: cg/meta/clean/demultiplexed_flow_cells.py [hunk 2]
```diff
@@ -136,13 +148,11 @@ def files_exist_on_disk(self) -> bool:
...
+                    self._files_exist_on_disk: bool = all(fastq_files_exist_on_disk)
                 if self.spring_files_exist_in_housekeeper:
-                    self._files_exist_on_disk: bool = all(
-                        [Path(spring_file.path).exists() for spring_file in self.hk_spring_files]
-                    )
+                    spring_files_exist_on_disk = self._check_files_existence(self.hk_spring_files)
```
Comment (henrikstranneheim) [inline cg/meta/clean/demultiplexed_flow_cells.py:154]:
```suggestion
                    spring_files_exist_on_disk: List[bool] = self._check_files_existence(self.hk_spring_files)
```

**Review by henrikstranneheim — APPROVED** [review]
👍

**Comment by barrystokman** [comment]
image

**Pull request merged**